### PR TITLE
Bug: Don't create events when blob lease cannot be acquired

### PR DESCRIPTION
**Pull request by rkondratowicz**
And don't log an error, just skip that file.

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java [hunk 1]
```diff
@@ -54,8 +56,8 @@
             )
         );
 
-    @Mock BlobClient blobClient;
-    @Mock BlobProperties blobProperties;
+    @Mock(lenient = true) BlobClient blobClient;
+    @Mock(lenient = true) BlobProperties blobProperties;
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: `blobExists` method mock all the properties of the blob. Not all are used before the lease is acquired

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java [hunk 2]
```diff
@@ -220,6 +222,24 @@ void should_not_upload_to_crime_when_blob_does_not_contain_envelope_entry() {
         verify(envelopeService).saveEvent(SOURCE_CONTAINER, "envelope1.zip", Event.FILE_PROCESSING_STARTED);
     }
 
+    @Test
+    void should_not_create_events_when_lease_cant_be_acquired() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: should_not_create_events_when_lease_cannot_be_acquired ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java [hunk 3]
```diff
@@ -63,14 +63,17 @@ public void process(BlobClient blobClient) {
 
         logger.info("Processing {} from {} container", blobName, containerName);
 
-        envelopeService.saveEvent(containerName, blobName, Event.FILE_PROCESSING_STARTED);
+        BlobLeaseClient leaseClient = leaseClientProvider.get(blobClient);
+        try {
+            leaseClient.acquireLease(60);
+        } catch (Exception exc) {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java]: Maybe we should catch some specific exception, that represents a conflict or sth? We still might have problems connecting and that should result in logging an error.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java]: done

**Pull request merged**